data/google-tag-manager.json

```diff
@@ -15,9 +15,17 @@
         "key": "gtm"
       },
       {
-        "code": "window[{{l}}]=window[{{l}}]||[];window[{{l}}].push({'gtm.start':new Date().getTime(),event:'gtm.js'});",
+        "code": "window[{{l}}]=window[{{l}}]||[];(function () {window[{{l}}].push(arguments)})('consent', {{consentType}},{{consentValues}});window[{{l}}].push({'gtm.start':new Date().getTime(),event:'gtm.js'});",
```

Comment:
Are we certain that including this in every Analytics `gtag` snippet is safe (i.e. leads to the same behavior as if it wasn't included)? I wonder because it's not mentioned in the default Analytics snippet.

Without being certain about this, I'd feel more comfortable if we could make inclusion of this call conditional, e.g. only include the `consent` call if the `consentType` and `consentValue` argument are specified/not empty, and then leave the defaults empty and up to the consumer.

Reply:
hmm yeah, that's a valid point. 
Could we use one optionalParam ("consent") and default it to null?
In that way it wouldn't get included unless you pass in an object?

Reply:
@housseindjirdeh @huang-julien

Reply:
default everything to null ? What would happen if we call gtm with nothing ? Will it ignore it ?

Reply:
It's not required to provide any `consent` to GTM, so not making the `consent` call would work fine. For instance, the default GTM snippets don't include that portion either (see https://support.google.com/tagmanager/answer/14847097 for example).

Reply:
I guess we probably want to have `(function () {window[{{l}}].push(arguments)})('consent', {{consentType}},{{consentValues}})` conditionnally. 

The best thing to do is probbly to provide a `preConfigCode` optionnal param so user will be even more free to add what they want. THe only issue is that `formatCode` stringify strings before insertion so we won't be able to add code with replacement tokens.

Reply:
I think it'd be safer to avoid direct code injection, so allowlisting certain kinds of parameters would IMO be a better option. I think the current approach of conditional `consentType` and `consentValues` works well, or alternatively we could use just a single `consent` key which includes both (e.g. in an array).

There aren't many of those special use-cases where configuration is needed _prior_ to the config line. The only other one I'm aware of is the `gtag('set', ...)` command (see #68, related: #71), which we could also support with an optional parameter (let's leave that for a separate issue/PR though).

Reply:
Related: With the approach from #73, we would be able to conditionally pass the consent call here too, leaving the defaults as it is today.